test/e2e/network_policy.go

```diff
@@ -0,0 +1,509 @@
+						NamespaceSelector: &unversioned.LabelSelector{
+							MatchLabels: map[string]string{
+								"ns-name": nsBName,
+							},
+						},
+					}},
+				}},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			By("Cleaning up the policy")
+			if err := f.Client.NetworkPolicies(nsA.Name).Delete(policy.Name, nil); err != nil {
+				framework.Failf("unable to delete policy %v: %v", policy.Name, err)
+			}
+		}()
+
+		// Create a pod with name 'client-a', which should be able to communicate with server.
+		By("Creating client-a in ns-a which should *not* be able to contact the server.")
+		podClientA := createClientPod(f, nsA, clientAName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod client-a")
+			if err := f.Client.Pods(nsA.Name).Delete(podClientA.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientA.Name, err)
+			}
+		}()
+
+		// Create a pod with name 'client-b', which will attempt to comunicate with the server,
+		//  but should not be able to do to the label-based policy.
+		By("Creating client-b in ns-b which should be able to contact the server.")
+		podClientB := createClientPod(f, nsB, clientBName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod client B")
+			if err := f.Client.Pods(nsB.Name).Delete(podClientB.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientB.Name, err)
+			}
+		}()
+
+		framework.Logf("Waiting for both clients to run.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientA)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-A to run")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientB)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-B to run")
+
+		framework.Logf("Waiting for client-a to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientA.Name, nsA.Name)
+		Expect(err).To(HaveOccurred(), "checking Client-A could not communicate with server.")
+
+		framework.Logf("Waiting for client-b to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientB.Name, nsB.Name)
+		Expect(err).NotTo(HaveOccurred(), "checking Client-B could communicate with server")
+	})
+})
+
+// Create a server pod with a listening container for each port in ports[].
+// Will also assign a pod label with key: "pod-name" and label set to the given podname for later use by the network
+// policy.
+func createServerPod(f *framework.Framework, namespace *api.Namespace, podName string, ports []int) (*api.Pod, *api.Service) {
+	// Because we have a variable amount of ports, we'll first loop through and generate our Containers for our pod,
+	// and ServicePorts.for our Service.
+	containers := []api.Container{}
+	servicePorts := []api.ServicePort{}
+	for i, port := range ports {
+		// Create the Container
+		containers = append(containers, api.Container{
+			Name:  fmt.Sprintf("%s-container-%d", podName, i),
```

Comment:
Instead of naming this after `i`, could you name it after the port it is listening on?